hypershift-operator/controllers/nodepool/manifests.go

```diff
@@ -150,3 +151,24 @@ func TokenSecret(namespace, name, payloadInputHash string) *corev1.Secret {
 		},
 	}
 }
+
+func AgentMachineTemplate(nodePool *hyperv1.NodePool, controlPlaneNamespace string) *agentv1.AgentMachineTemplate {
+	agentMachineTemplate := &agentv1.AgentMachineTemplate{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Annotations: map[string]string{
+				nodePoolAnnotation: ctrlclient.ObjectKeyFromObject(nodePool).String(),
+			},
+			Namespace: controlPlaneNamespace,
+		},
+		Spec: agentv1.AgentMachineTemplateSpec{
+			Template: agentv1.AgentMachineTemplateResource{
+				Spec: agentv1.AgentMachineSpec{},
+			},
+		},
+	}
+	specHash := hashStruct(agentMachineTemplate.Spec.Template.Spec)
```

Comment:
This means that if the spec ever gets changed (Doesn't happen today but might in the future?) we will create a new template, is that intended?

Reply:
I copied this from AWSMachineTemplate in the same file, I assumed this is how it was intended in HyperShift

Reply:
Any change to the nodePool platform specifics it's propagated and result in a new template being created so rolling upgrades can be supportd. This is by design atm. Template cleanup needs to be implemented in a platform agnostic way aside from this PR https://github.com/openshift/hypershift/pull/760#discussion_r765778818